### DISCUSSION TITLE
Teleportation (Moving)  the Body with Constraints

### DISCUSSION THREAD

**Discussion by Logris** [jrouwe/JoltPhysics#1396] [Q&A] [answered]
Hi, 
There is a body **A** (main body).  Bodies **B, C, D** ... are attached to it...
The attached bodies can also be composite, for example, a chain.

How can I move body A to a new world position with all connected bodies?

`SetPositionAndRotation(...)` for body A causes the bodies to explode and may cause Jolt to crash.

**Comment by jrouwe** [comment, marked as answer]
You would have to call `BodyInterface::SetPositionAndRotation` on B, C and D too and place them so that their relative position/orientation to A is unchanged. I would also recommend doing a collision check to see if the space that you're moving A .. D to is not occupied.
